src/app/core/api.service.js

```diff
@@ -0,0 +1,39 @@
+(function() {
+    'use strict';
+
+    angular
+        .module('xyz-cv-ui.core')
+        .factory('api', api);
+
+    function api($resource, config) {
+        var service = {
+            post: post,
+            query: query,
+            get: get,
+            update: update,
+            remove: remove
+        };
+        return service;
+
+
+        function post(path, object) {
+            return $resource(config.API_URL + path).save(object);
+        }
+
+        function query(path) {
+            return $resource(config.API_URL + path).query();
+        }
+
+        function get(path, id) {
+            return $resource(config.API_URL + path + '/:id', {id: id}).get();
+        }
+
+        function update(path, id, object) {
```

Comment:
Yesterday I found this, and I liked it :) http://kirkbushell.me/angular-js-using-ng-resource-in-a-more-restful-manner/
That way you can always just call `save` and it'll do the right thing :) See if you can implement that!